core/zoom_controls.js

```diff
@@ -302,7 +302,11 @@ Blockly.ZoomControls.prototype.createZoomResetSvg_ = function(rnd) {
   Blockly.bindEventWithChecks_(zoomresetSvg, 'mousedown', null, function(e) {
     ws.markFocused();
     ws.setScale(ws.options.zoomOptions.startScale);
+    ws.beginCanvasTransition();
     ws.scrollCenter();
+    setTimeout(function() {
+      ws.endCanvasTransition();
+    }, 500);
```

Comment:
How does this behave in a race condition where the user clicks the button twice in less than half a second?  Change the timings to 10 seconds to see what happens.

You might need to keep the PID generated by setTimeout in a variable, and clear the timeout if another call is made.

Reply:
It shouldn't ever end in a bad state, since it's not relying on the PID to know when to end the canvas transition.  So the worst case is that it jumps too fast, and that should hardly be noticeable.  The end state will be the correct CSS.